internal/wrappers/feature-flags.go

```diff
@@ -55,26 +58,70 @@ var FeatureFlagsBaseMap = []CommandFlags{
 }
 
 var FeatureFlags = map[string]bool{}
+var FeatureFlagsSpecific = map[string]bool{}
 
 func HandleFeatureFlags(featureFlagsWrapper FeatureFlagsWrapper) error {
 	allFlags, err := featureFlagsWrapper.GetAll()
 	if err != nil {
-		loadFeatureFlagsDefaultValues()
-
+		LoadFeatureFlagsDefaultValues()
 		return nil
 	}
 
 	loadFeatureFlagsMap(*allFlags)
-
 	return nil
 }
 
+func GetSpecificFeatureFlag(featureFlagsWrapper FeatureFlagsWrapper, flagName string) (*FeatureFlagResponseModel, error) {
+	if value, exists := FeatureFlagsSpecific[flagName]; exists {
+		return &FeatureFlagResponseModel{Name: flagName, Status: value}, nil
+	}
+
+	specificFlag, err := getSpecificFlagWithRetry(featureFlagsWrapper, flagName, maxRetries)
+	if err != nil {
+		UpdateSpecificFeatureFlagMapWithDefault(flagName)
+		return &FeatureFlagResponseModel{Name: flagName, Status: FeatureFlagsSpecific[flagName]}, nil
+	}
+
+	UpdateSpecificFeatureFlagMap(flagName, *specificFlag)
+	return specificFlag, nil
+}
+
+func getSpecificFlagWithRetry(wrapper FeatureFlagsWrapper, flagName string, retries int) (*FeatureFlagResponseModel, error) {
+	var flag *FeatureFlagResponseModel
+	var err error
+
+	for i := 0; i < retries; i++ {
+		flag, err = wrapper.GetSpecificFlag(flagName)
+		if err == nil {
+			return flag, nil
+		}
+	}
+
+	return nil, errors.New("failed to get feature flag after retries")
+}
+
+func UpdateSpecificFeatureFlagMapWithDefault(flagName string) {
+	logger.PrintIfVerbose("Get feature flags failed. Loading defaults...")
```

Comment:
I think the log suppose to be when it failed and not in the update method.. you can remove it and follow other comments

Reply:
We don't want to know that the value is currently the default?
Is it because the API call didn't return anything/didn't work?

Reply:
UpdateSpecificFeatureFlagMapWithDefault is delete